components/constellation/constellation.rs

```diff
@@ -2733,6 +2740,13 @@ where
         if let Err(e) = storage_ipc_receiver.recv() {
             warn!("Exit storage thread failed ({:?})", e);
         }
+        if self.webgl_threads.is_some() {
+            if let Some(webgl_threads_receiver) = webgl_threads_receiver {
+                if let Err(e) = webgl_threads_receiver.recv() {
+                    warn!("Exit WebGL thread failed ({:?})", e);
+                }
+            }
+        }
```

Comment:
Is it impossible `webgl_threads_receiver` to be `Some(...)` when `webgl_threads` is None? If so, this should probably be an assertion instead. If not, matching a tuple is might be bit cleaner here. Either way, the error can be inline in the format statement.

```suggestion
        if let Some(webgl_threads_receiver) = webgl_threads_receiver {
            assert!(self.webgl_threads.is_some());
            if let Err(e) = webgl_threads_receiver.recv() {
                warn!("Exit WebGL thread failed ({e:?})");
            }
        }
```

```suggestion
        match (self.webgl_threads, webgl_threads_receiver)  {
            (Some(_), Some(webgl_threads_receiver)) => if let Err(e) = webgl_threads_receiver.recv() {
                    warn!("Exit WebGL thread failed ({e:?})");
             },
            _ => {},
        }
```

Reply:
indeed tuple matching is better.